jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/grid/TreeDataGrid.java

```diff
@@ -421,4 +425,36 @@ protected void onAfterApplyColumnSecurity(AbstractGridDelegate.ColumnSecurityCon
             super.removeColumn(context.getColumn());
         }
     }
+
+    @Override
+    public JmixGridContextMenu<E> getContextMenu() {
+        if (contextMenu == null) {
+            contextMenu = new JmixGridContextMenu<>();
+            contextMenu.setTarget(this);
+        }
+        return contextMenu;
+    }
+
+    @Override
+    public GridContextMenu<E> addContextMenu() {
+        throw new UnsupportedOperationException("TreeDataGrid can have only one context menu attached, " +
```

Comment:
It's better not to hardcode component name, `getClass().getSimpleName()` is preferable